src/Backups/BackupsWorker.cpp

```diff
@@ -449,9 +451,10 @@ void BackupsWorker::doBackup(
         backup.reset();
 
         LOG_INFO(log, "{} {} was created successfully", (backup_settings.internal ? "Internal backup" : "Backup"), backup_name_for_logging);
-        setStatus(backup_id, BackupStatus::BACKUP_CREATED);
         /// NOTE: we need to update metadata again after backup->finalizeWriting(), because backup metadata is written there.
         setNumFilesAndSize(backup_id, num_files, total_size, num_entries, uncompressed_size, compressed_size, 0, 0);
```

Comment:
setNumFilesAndSize() doesn't update the backup log, does it? It's better to have a comment about that.

Reply:
Each backup/restore operation creates two entries in the `backup_log`:
1. When the operation starts, it records start `date` and `time (us)` of the log entry itself, as well ad `id`, `name`, initial `status` and `start_time` of the operation.
2. When the operation ends, it records the same fields plus `end_time` and either `num_files`, `total_size`, `num_entries`, `uncompressed_size`, `compressed_size`, `files_read`, `bytes_read` in case of success of `error` in case of failure.
The `setNumFilesAndSize()` method updates fields of `BackupOperationInfo` `num_files`, `total_size`, `num_entries`, `uncompressed_size`, `compressed_size`, `num_read_files` and `num_read_bytes` of current operation's entry in the `infos` map, which in turn is inserted as the second entry after the operation has finished.
In a nutshell, `setNumFilesAndSize()` updates the current operation's entry in the `infos` map, and `setStatus()` flushes it to the log as the second line.
Probably, in the future we will come up (maybe not) with some intermediate steps to reflect the progress of long operations.

Reply:
I meant the function `setNumFilesAndSize()` could call

```
    if (backup_log)
        backup_log->add(BackupLogElement{info});
```

just like `setStatus()` does.

Reply:
Sure thing, got it.
As I developed this feature, I also thought about adding a record to `backup_log` within `setNumFilesAndSize()` method, but eventually decided to do it immediately in a subsequent `setStatus()` call which appears to be inevitable regardless of the final operation status - failure or success.
Done (I placed the comment at the beginning of the `setNumFilesAndSize()` method).